gcp_variant_transforms/libs/annotation/vep/vep_runner.py

```diff
@@ -0,0 +1,464 @@
+      vep_num_fork,  # type: int
+      pipeline_args  # List[str]
+      ):
+    # type: (...) -> None
+    """Constructs an instance for running VEP.
+
+    Note that external users of this class can use create_runner_and_update_args
+    function of this module to create an instance of this class from flags.
+
+    Args:
+      input_pattern: The pattern to identify all input files.
+      output_dir: The location for all output files. This is expected not to
+        exist and is created in the process of running VEP pipelines.
+      vep_image_uri: The URI of the image that contains VEP.
+      vep_cache_path: The URI for the cache file on GCS.
+      vep_num_fork: The value of the --fork argument for running VEP.
+      pipeline_args: The list of arguments that are meant to be used when
+        running Beam; for simplicity we use the same arguments to decide how
+        many and what type of workers to use, where to run, etc.
+    """
+    self._pipeline_service = pipeline_service
+    self._vep_image_uri = vep_image_uri
+    self._vep_cache_path = vep_cache_path
+    self._vep_num_fork = vep_num_fork
+    self._input_pattern = input_pattern
+    self._output_dir = output_dir
+    self._vep_info_field = vep_info_field
+    self._process_pipeline_args(pipeline_args)
+    # _current_operations holds the list of operation names currently running.
+    self._running_operation_ids = []  # type: List[str]
+
+  def get_output_pattern(self):
+    # type: () -> str
+    return (self._format_dir_path(self._output_dir) +
+            _GCS_RECURSIVE_WILDCARD + _VEP_OUTPUT_SUFFIX)
+
+  def _get_api_request_fixed_parts(self):
+    # type: () -> Dict
+    """Returns the part of API request that is fixed between actions.
+
+    This includes setting up VEP cache, virtual machine setup, etc. The variant
+    parts are the `commands` for processing each file and should be added before
+    sending the API request.
+    """
+    return {
+        _API_PIPELINE: {
+            _API_ACTIONS: [
+                self._make_action('mkdir', '-p', '/mnt/vep/vep_cache'),
+                self._make_action('gsutil', '-q', 'cp', self._vep_cache_path,
+                                  '/mnt/vep/vep_cache/')
+            ],
+            'environment': {
+                'VEP_CACHE': '/mnt/vep/vep_cache/{}'.format(
+                    _get_base_name(self._vep_cache_path)),
+                'NUM_FORKS': str(self._vep_num_fork),
+                'VCF_INFO_FILED': self._vep_info_field,
+                # TODO(bashir2): Decide how to do proper reference validation,
+                # the following --check_ref just drops variants that have
+                # wrong REF. If there are too many of them, it indicates that
+                # VEP database for a wrong reference sequence is being used
+                # and this has to caught and communicated to the user.
+                'OTHER_VEP_OPTS':
+                    '--everything --check_ref --allow_non_variant',
```

Comment:
Let's leave as is for this PR. I agree with you that limiting the # of flags is a good idea. In general, users get confused and will use whatever is the default values anyway.